torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+
+    The dataset contains 4 classes:
+    0: paddy rice
+    1: maize
+    2: soybean
+    3: others
+
+
+    Dataset format:
+    1) Three .TIF files containing the labels
+    2) JavaScript code to download images from the dataset.
+
+    If you use this dataset in your research, please use the corresponding citation:
+    * You, N., Dong, J., Huang, J. et al. The 10-m crop type maps in
+    Northeast China during 2017-2019.
+    Sci Data 8, 41 (2021). https://doi.org/10.1038/s41597-021-00827-9
+    """
+
+    filename_regex = r"CDL(?P<year>\d{4})_clip"
+    filename_glob = "CDL*.tif"
+    # there is only a single zipfolder, none for the files
+    zipfile_glob = "13090442.zip"
+
+    date_format = "%Y"
+    is_image = False
+
+    # need to add url
+    url = "https://figshare.com/ndownloader/articles/13090442/versions/1"
+
+    # md5s = {
+    #     2017: "a85341fa6248fd7e0badab6c",
+    #     2018: "478d10786aa798fb11693ec1",
+    #     2019: "42e483fdc8239d22dba7020f"
+    # }
+
+    md5s = {"main": "eae952f1b346d7e649d027e8139a76f5"}
+
+    years = [2017, 2018, 2019]
+
+    cmap = {
+        0: (0, 0, 0, 0),
+        1: (0, 255, 0, 255),
+        2: (255, 0, 0, 255),
+        3: (255, 255, 0, 255),
+        4: (255, 0, 255, 255),
+    }
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        years: list[int] = [2019],
+        classes: list[int] = list(cmap.keys()),
+        transforms: Optional[Callable[[dict[str, Any]], dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new dataset.
+
+        Args:
+            paths: one or more root directories to search or files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            years: list of years for which to use dataset
+            classes: list of classes to include, the rest will be mapped to 0
+                (defaults to all classes)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            AssertionError: if ``years`` or ``classes`` are invalid
+            FileNotFoundError: if no files are found in ``paths``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        assert all(
+            year in self.years for year in years
```

Comment:
Might be simpler to use `set(years) <= set(self.years)`. But only if we decide to keep the `years` parameter